packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol

```diff
@@ -153,29 +156,32 @@ contract FuelERC20GatewayV4 is
      * @param _amount The amount to reset the limit to.
      * @param _rateLimitDuration The new rate limit duration.
      * Fuel's implementation is inspired by the Linea Bridge dessign (https://github.com/Consensys/linea-contracts/blob/main/contracts/messageService/lib/RateLimiter.sol)
-     * Only point of difference from the linea implementation is that when currentPeriodEnd >= block.timestamp then if the new rate limit amount is less than the currentPeriodAmount, then currentPeriodAmount is not updated this makes sure that if rate limit is first reduced & then increased within the rate limit duration then any extra amount can't be withdrawn
+     * Only point of difference from the linea implementation is that rate limit params are updated unconditionally
      */
-    function resetRateLimitAmount(address _token, uint256 _amount, uint256 _rateLimitDuration) external onlyRole(SET_RATE_LIMITER_ROLE) {   
-        // avoid multiple SLOADS
-        uint256 rateLimitDurationEndTimestamp = currentPeriodEnd[_token];
-        
+    function resetRateLimitAmount(address _token, uint256 _amount, uint256 _rateLimitDuration) external onlyRole(SET_RATE_LIMITER_ROLE) {           
         // set new rate limit duration
         rateLimitDuration[_token] = _rateLimitDuration;
         
-        // if period has expired then currentPeriodAmount is zero
-        if (rateLimitDurationEndTimestamp < block.timestamp) {
-            unchecked {
-                currentPeriodEnd[_token] = block.timestamp + _rateLimitDuration;
-            }
-
-            currentPeriodAmount[_token] = 0;
+        unchecked {
+            currentPeriodEnd[_token] = block.timestamp + _rateLimitDuration;
         }
 
+        currentPeriodAmount[_token] = 0;
+
         limitAmount[_token] = _amount;
 
         emit RateLimitUpdated(_token, _amount);
     }
 
+    /**
+     * @notice Disables/Re-Enables rate limit.
+     * @param _token The token address to disable rate limit for.
+     * @param _disableRateLimit flag to disable rate limit or re-enable rate limit.
+     */
+    function disableRateLimit(address _token, bool _disableRateLimit) external onlyRole(SET_RATE_LIMITER_ROLE) {
```

Comment:
I think this function name is also confusing. Let's rename it to something like `setWhateverLimit`